core/src/main/java/org/apache/calcite/sql/fun/SqlDateFunction.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to you under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.calcite.sql.fun;
+
+import org.apache.calcite.sql.SqlFunction;
+import org.apache.calcite.sql.SqlFunctionCategory;
+import org.apache.calcite.sql.SqlKind;
+import org.apache.calcite.sql.SqlOperatorBinding;
+import org.apache.calcite.sql.SqlSyntax;
+import org.apache.calcite.sql.type.OperandTypes;
+import org.apache.calcite.sql.type.ReturnTypes;
+import org.apache.calcite.sql.validate.SqlMonotonicity;
+
+/**
+ * The <code>DATE</code> function.
+ */
+public class SqlDateFunction extends SqlFunction {
+  //~ Constructors -----------------------------------------------------------
+
+  public SqlDateFunction() {
+    super(
+        "DATE",
+        SqlKind.OTHER_FUNCTION,
+        ReturnTypes.DATE,
+        null,
+        OperandTypes.NILADIC,
+        SqlFunctionCategory.TIMEDATE);
+  }
+
+  //~ Methods ----------------------------------------------------------------
+
+  public SqlSyntax getSyntax() {
+    return SqlSyntax.FUNCTION_ID;
+  }
+
+  @Override public SqlMonotonicity getMonotonicity(SqlOperatorBinding call) {
+    return SqlMonotonicity.INCREASING;
+  }
+
+  // Plans referencing context variables should never be cached
+  public boolean isDynamicFunction() {
```

Comment:
Can we just omit this function altogether and let it use the base class version?

Reply:
I'm not too sure the significance of this but it seems that it is false by default.

Reply:
I have it here in the first place as I was referencing [SqlCurrentDateFunction](https://github.com/googleinterns/calcite/blob/master/core/src/main/java/org/apache/calcite/sql/fun/SqlCurrentDateFunction.java)

Reply:
Ok sounds good. Either way seems fine.